src/meta/src/rpc/service/ddl_service.rs

```diff
@@ -233,6 +237,156 @@ where
         }))
     }
 
+    async fn create_sink(
+        &self,
+        request: Request<CreateSinkRequest>,
+    ) -> Result<Response<CreateSinkResponse>, Status> {
+        self.ddl_lock.read().await;
+        self.env.idle_manager().record_activity();
+
+        let req = request.into_inner();
+        let mut mview = req.get_sink().map_err(tonic_err)?.clone();
+        let fragment_graph = req.get_fragment_graph().map_err(tonic_err)?.clone();
+        // 0. Generate an id from mview.
+        let id = self
+            .env
+            .id_gen_manager()
+            .generate::<{ IdCategory::Table }>()
+            .await
+            .map_err(tonic_err)? as u32;
+        mview.id = id;
+
+        // 1. Resolve the dependent relations.
```

Comment:
Can't we reuse the code instead of duplicating them?

Reply:
I have introduced `fn get_dependent_relations(&self, fragment_graph: &StreamFragmentGraph) -> RwResult<Vec<u32>>`  method to reduce code duplication in `fn create_sink` and `fn create_materialized_view`.